### PR TITLE
add boost build generator

### PR DESCRIPTION
**Pull request by arvidn**
This makes it possible to consume conan packages from a Jamfile with boost-build. I've tested it in a package form here: https://github.com/arvidn/conan-boost-build-gen

### CODE REVIEW

**Comment by skizzay** [comment]
I think this is a very useful generator to have available built in.  Two suggestions:
1. Rename it to ```boost-jam```.  I feel that we can run into confusion regarding a Boost build (a compilation of Boost) rather than the Boost system.  I feel that ```boost-jam``` would remove any naming ambiguities.
2. Can we add a section to the docs in the Generators section? :)

**Comment by memsharded** [comment]
Thanks for the contribution @arvidn !!

What do you think about the @skizzay suggestion of the renaming?
I would also suggest adding some tests. No need to be complete or run a real build, just unit tests to check that things are in place. Please tell me if you can add them, otherwise, I'll contribute them.

File: conans/client/generators/boostbuild.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+from conans.model import Generator
+from conans import ConanFile
+
+def JamfileOutput(dep_cpp_info):
+    out = ''
+    for lib in dep_cpp_info.libs:
+        out += 'lib %s :\n' % lib
+        out += '\t: # requirements\n'
+        out += '\t<name>%s\n' % lib
+        out += ''.join('\t<search>%s\n' % x for x in dep_cpp_info.lib_paths)
```
Comment (lasote) [inline conans/client/generators/boostbuild.py]: For Windows paths, the \\ should be scaped, use:
`from conans.client.files import unix_path`
`unix_path(x) for x in dep_cpp_info.lib_paths`
Apply the same for include_paths

**Comment by arvidn** [comment]
thanks for the feedback. I've been kind of busy since cppcon. I hope to pick this up again next week or so.

File: conans/client/generators/boostbuild.py (hunk 1 above)
Comment (memsharded) [reply conans/client/generators/boostbuild.py]: Beware that ``unix_path`` might not work in all contexts, I don't think it is the good solution here

Comment (lasote) [reply conans/client/generators/boostbuild.py]: Then we should `x.replace("\\", "/") for x in dep_cpp_info.lib_paths`

**Comment by lasote** [comment]
I'll open a PR for you. We are trying to release 0.28 tomorrow and would be nice if we release this and people can start trying it.

File: conans/client/generators/boostbuild.py (hunk 1 above)
Comment (arvidn) [reply conans/client/generators/boostbuild.py]: the path conan passes in, is it always a windows path? Windows paths do work in boost-build on windows, but the backslashes need to be escaped. I would have expected the path from conan to be in the native format, I'm not 100% sure unix paths always work on windows here.

A related issue, if there are spaces in the path, there needs to be quotes around it. It probably makes sense to always quote these paths.

**Comment by memsharded** [comment]
Ok, so the most important thing here is of course the name. I agree it is better to change it, I would suggest something like ``conanbuildinfo.jam``, because that would be aligned with other generators. Other name could be ``conan-deps-config.jam``. What do you think @arvidn? Given you contributed it, would you like to change it yourself? In any case I would try to add it to 0.30 release next week, wouldn't like to break after the 1.0 release.

**Comment by arvidn** [comment]
I don't actually know how to (or if it's possible to) include a specific file in a Jamfile, other than with ``use-project``, but that pulls in a file called ``Jamfile`` from a specific directory.

I suppose the generator could produce a directory called ``conandeps`` (or something like it) and put a ``Jamfile`` in there. then it could be referred to by ``use-project /deps : conandeps ;``

**Pull request merged**